src/NzbDrone.Core/MediaFiles/EpisodeFileMovingService.cs

```diff
@@ -1,85 +1,85 @@
+                                IUpdateMediaInfo updateMediaInfo,
                                 IBuildFileNames buildFileNames,
                                 IDiskTransferService diskTransferService,
                                 IDiskProvider diskProvider,
                                 IMediaFileAttributeService mediaFileAttributeService,
+                                IScriptImportDecider scriptImportDecider,
+                                IRenameEpisodeFiles renameEpisodeFiles,
                                 IEventAggregator eventAggregator,
                                 IConfigService configService,
                                 Logger logger)
+            : base(updateEpisodeFileService, buildFileNames, diskProvider, mediaFileAttributeService, eventAggregator, logger)
         {
-            _episodeService = episodeService;
-            _updateEpisodeFileService = updateEpisodeFileService;
-            _buildFileNames = buildFileNames;
+            _updateMediaInfo = updateMediaInfo;
             _diskTransferService = diskTransferService;
-            _diskProvider = diskProvider;
-            _mediaFileAttributeService = mediaFileAttributeService;
-            _eventAggregator = eventAggregator;
+            _scriptImportDecider = scriptImportDecider;
+            _renameEpisodeFiles = renameEpisodeFiles;
             _configService = configService;
             _logger = logger;
         }
 
-        public EpisodeFile MoveEpisodeFile(EpisodeFile episodeFile, Series series)
+        protected override void ExecuteTransfer(string episodeFilePath, string destinationFilePath, TransferMode mode, ScriptImportDecisionInfo scriptImportDecisionInfo)
         {
-            var episodes = _episodeService.GetEpisodesByFileId(episodeFile.Id);
-            var filePath = _buildFileNames.BuildFilePath(episodes, series, episodeFile, Path.GetExtension(episodeFile.RelativePath));
+            scriptImportDecisionInfo.mode = mode;
+            var episodeFile = scriptImportDecisionInfo.episodeFile;
+            var series = scriptImportDecisionInfo.localEpisode.Series;
 
-            EnsureEpisodeFolder(episodeFile, series, episodes.Select(v => v.SeasonNumber).First(), filePath);
+            var scriptImportDecision = _scriptImportDecider.TryImport(episodeFilePath, destinationFilePath, scriptImportDecisionInfo);
 
-            _logger.Debug("Renaming episode file: {0} to {1}", episodeFile, filePath);
-
-            return TransferFile(episodeFile, series, episodes, filePath, TransferMode.Move);
+            switch (scriptImportDecision)
+            {
+                case ScriptImportDecision.DeferMove:
+                    _diskTransferService.TransferFile(episodeFilePath, destinationFilePath, mode);
+                    break;
+                case ScriptImportDecision.RenameRequested:
+                    _updateMediaInfo.Update(episodeFile, series, false);
```

Comment:
Why would the script import the file and then immediately need it to be renamed and moved? If it needs to be moved why can't the script do that and communicate the new file name or instead of moving it, leave it and let Sonarr handle it?

Reply:
Well the issue is that Sonarr's naming is dynamic and can be pretty involved, so if you still want the ability to use Sonarr's rename, you'd have to replicate that perfectly in the script, which would be rather impractical.

Reply:
Ahh gotcha, in that case I wonder if it's more practical to have Sonarr process the file again the next pass (but that could be problematic with multiple files on disk) or just recalculate the filename (after updating mediainfo). Right now forcing a move would be problematic for seeding torrents, even if the file actually hadn't been altered.

Reply:
I don't think it's an issue actually. The move is from the imported file, that's why I'm setting the Path to null there, then it gets recalculated to the path inside the library.